sklego/preprocessing/dictmapper.py

```diff
@@ -0,0 +1,75 @@
+import numpy as np
+from sklearn.base import BaseEstimator, TransformerMixin
+from sklearn.utils import check_array
+from sklearn.utils.validation import check_is_fitted
+from functools import partial
+
+
+class DictMapper(TransformerMixin, BaseEstimator):
+    """
+    Map the values of values of columns according to the input dictionary,
+    fall back to the default if the key is not present in the dictionary.
+
+    :param mapper: The dictionary containing the mapping of the values
+    :param default: The value to fall back to if the value is not in the mapper
+    """
+
+    def __init__(self, mapper, default):
+        self.mapper = mapper
+        self.default = default
+
+    def fit(self, X, y=None):
+        """
+        Checks the input dataframe and records the shape of it
+
+        :type X: pandas.DataFrame or numpy.ndarray
+        :param X: The column(s) from which the mapping will be applied
+
+        :param y: Ignored.
+
+        :rtype: sklego.preprocessing.DictMapper
+        :returns: The fitted object.
+
+        :raises:
```

Comment:
Do we raise a ValueError if X contains a non-numeric column here? This might have ben a copy and paste too much.

Reply:
I think this is my final comment.

Reply:
My bad, was still working on it